### PR TITLE
Refactor mechs into a new obj, /obj/exosuit

### PR DESCRIPTION
**Pull request by ghost**
- [x] Base /obj/exosuit movement
- [ ] climbing into exosuits
- [ ] equipment
- [ ] actually refactor mechs

:cl: More Robust Than You
refactor: Refactored mechs
/:cl:

### CODE REVIEW

**Comment by Bawhoppen** [comment]
he didn't want to make space pods mechs and he stuck to his guns

**Comment by kevinz000** [comment]
please dont fuck this up

File: code/__DEFINES/flags.dm [hunk 1]
```diff
@@ -93,3 +93,18 @@ GLOBAL_LIST_INIT(bitflags, list(1, 2, 4, 8, 16, 32, 64, 128, 256, 512, 1024, 204
 #define ACID_PROOF 32 //acid stuck on it doesn't melt it.
 #define INDESTRUCTIBLE 64 //doesn't take damage
 #define FREEZE_PROOF 128 //can't be frozen
+
+//Exosuit flags
+#define OCCUPANT_CAN_DRIVE	1
+#define OCCUPANT_PASSENGER	2
+
+#define MULTIMOVE		1
```
Comment (kevinz000) [inline code/__DEFINES/flags.dm]: why not TURFMOVE 1 SPACEMOVE 2 and have both be 3 because bitflags?

File: code/modules/exosuit/exosuit.dm [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		. = TRUE
+
+/obj/exosuit/proc/move_override(mob/user, direction)
+	return FALSE
+
+/obj/exosuit/proc/on_move(turf/T)
+	return FALSE
+
+/obj/exosuit/relaymove(mob/user, direction)
+	if(occupants[user] && occupants[user] != OCCUPANT_CAN_DRIVE)
```
Comment (kevinz000) [inline code/modules/exosuit/exosuit.dm]: not how you compare bitflags

**Comment by Qustinnus** [comment]
kevinz wtf u doing dont encourage him we're making our own mechs remember. smh. Some top 10 anime betrayal shit going on here

**Comment by Iamgoofball** [comment]
:-1:

vehicle code ffs

**Comment by MoreRobustThanYou** [comment]
@Iamgoofball STFU that's 4 bikes and the janitors autism mobile

**Comment by Iamgoofball** [comment]
you're retarded

vehicle code will let you remove all mecha movement copypaste

**Comment by MoreRobustThanYou** [comment]
@Iamgoofball I'm on mobile so I can't slam my head on my keyboard so hahahahahaha Bay f to jg f j do j can he go it go for I'm still here too too make end all for food to eat job so he got up he do

**Comment by Qustinnus** [comment]
having a common parent is better. there's only like 5 vehicles right now because the code for them is fucking trash.

**Comment by Incomptinence** [comment]
Stay stronk cat

**Comment by KorPhaeron** [comment]
Let me know when there is enough progress to be warrant a review and I'll reopen

**Pull request closed without merging**